cartographer_rviz/cartographer_rviz/submaps_display.h

```diff
@@ -32,6 +32,21 @@
 
 namespace cartographer_rviz {
 
+// This should be a private class in SubmapsDisplay, unfortunately, QT does not
+// allow for this.
+struct Trajectory : public QObject {
+  Q_OBJECT
+
+ public:
+  Trajectory(std::unique_ptr<::rviz::BoolProperty> property);
```

Comment:
Yeah, that part didn't confuse me, I added it in the first place :-) Even if you accept an rvalue reference in your prototype, it actually becomes an lvalue, so you still need the `std::move` cast. What confused me is that when @SirVer removed the rvalue reference from the variable property, this introduced an additional implicit move from the rvalue at the call site into the property variable.